### ISSUE TITLE
Multiple sticker thumbnail files with same content

### ISSUE THREAD

**Issue by zevlg** [tdlib/td#487]
Between restarts TDLib seems to download same sticker thumbnail file multiple times:

```console
$ find thumbnails/246115836_48082_* | xargs md5sum
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(0).webp
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(1).webp
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(2).webp
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(3).webp
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(4).webp
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(5).webp
62cb81be2e92e17eb8272cd3eda00c0f  thumbnails/246115836_48082_(6).webp
$
```

Is it okay, or am I doing something wrong?

Also I think I saw same behaviour for photos/avatas as well

**Comment by levlam** [comment]
Is there `246115836_48082.webp`? Is it TDLib 1.3.0?

**Comment by zevlg** [comment]
> Is there `246115836_48082.webp`? Is it TDLib 1.3.0?

File is there, yeah 1.3.0

**Comment by levlam** [comment]
Could you show modification time of the files?

**Comment by zevlg** [comment]
> Could you show modification time of the files?

```console
-rw-r----- 1 lg root 5686 янв 11 16:01 246115836_48082_(0).webp
-rw-r----- 1 lg root 5686 янв 11 16:17 246115836_48082_(1).webp
-rw-r----- 1 lg root 5686 янв 11 16:25 246115836_48082_(2).webp
-rw-r----- 1 lg root 5686 янв 15 16:54 246115836_48082_(3).webp
-rw-r----- 1 lg root 5686 янв 16 14:59 246115836_48082_(4).webp
-rw-r----- 1 lg root 5686 янв 17 15:05 246115836_48082_(5).webp
-rw-r----- 1 lg root 5686 фев  6 13:19 246115836_48082_(6).webp
-rw-r----- 1 lg root 5686 дек 17 13:30 246115836_48082.webp
```

Also, I just checked on my home laptop and got kind of similiar with profile_photos, but with some weird quotes (md5sum is the same):

```console
$ ls -al profile_photos/*7134*
-rw-r----- 1 lg root 10841 янв 16 22:43 '713423908_13675_(0).jpg'
-rw-r----- 1 lg root 10841 янв 16 22:19  713423908_13675.jpg
```

might be related

**Comment by levlam** [comment]
I've managed to reproduce this behavior for thumbnails if the previously downloaded file has been changed by external program, i.e. if its modification time has changed. In that case TDLib creates new file to save a content from an inline thumbnail.

Profile photos has no inline thumbnails, so this doesn't explain, how the second file has been created.

**Comment by nyuszika7h** [comment]
> with some weird quotes
That's just because on your laptop you have a newer version of coreutils (ls) where the default is to quote/shell-escape file names when necessary. The apostrophes aren't actually part of the file name.

**Comment by zevlg** [comment]
> > with some weird quotes
> 
> That's just because on your laptop you have a newer version of coreutils (ls) where the default is to quote/shell-escape file names when necessary. The apostrophes aren't actually part of the file name.

ah, nice catch, thanks!

I'll investigate deeper why modification time of some files changes, I have no such intentions in telegram client I'm writting

**Comment by levlam** [comment]
This was fixed in TDLib 1.5.0 for sticker thumbnails: new thumbnail file will not be created if modification time of the file was changed, but the content is still the same.
